torchtext/experimental/datasets/language_modeling.py

```diff
@@ -1,22 +1,38 @@
 import torch
-import logging
-import io
-from torchtext.utils import download_from_url, extract_archive
-from torchtext.vocab import build_vocab_from_iterator
 from torchtext.data.utils import get_tokenizer
-from torchtext.vocab import Vocab
+from torchtext.vocab import build_vocab_from_iterator
 from torchtext.data.functional import numericalize_tokens_from_iterator
+from torchtext.experimental.datasets import raw
 
-URLS = {
-    'WikiText2':
-        'https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-2-v1.zip',
-    'WikiText103':
-        'https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-103-v1.zip',
-    'PennTreebank':
-        ['https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.train.txt',
-         'https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.test.txt',
-         'https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.valid.txt']
-}
+
+def vocab_func(vocab):
```

Comment:
if `vocab_func`, `totensor`, `build_vocab` are same from those in `torchtext.experimental.datasets.text_classification`, we don't need to repeat here. I pull them to question answer datasets in the same way. https://github.com/pytorch/text/blob/edb143af512e890f3a00d3e872658f0986f3b23a/torchtext/experimental/datasets/question_answer.py#L5

Reply:
Or let's move them in a functional.py file similar to torchaudio and pytorch

Reply:
Duplicated functions `vocab_func`, `totensor` have been removed.